### PR TITLE
Add notification center to controller

### PR DESCRIPTION
**Pull request by kimikowang**
1. Notifiers are managed by the notification center. Each notifier is configured in `iter8config-notifiers` ConfigMap and can be added/updated /deleted while controller is running.

2. Now only slack integration is supported.

### CODE REVIEW

File: install/helm/iter8-controller/templates/metrics/iter8_metrics.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 kind: ConfigMap
 metadata:
-  name: iter8-metrics
+  name: iter8config-metrics
```
Comment (kalantar) [inline install/helm/iter8-controller/templates/metrics/iter8_metrics.yaml]: Why are we changing the name of the config map?

Comment (kalantar) [reply install/helm/iter8-controller/templates/metrics/iter8_metrics.yaml]: I see. Consistency with `iter8config-notifiers`. Have we considered having all of the config in the same configmap?

Comment (kimikowang) [reply install/helm/iter8-controller/templates/metrics/iter8_metrics.yaml]: Having them all in the same file might not be a good idea. Data in each ConfigMap is a yaml file indeed. They have different format structures.

File: pkg/controller/experiment/common.go [hunk 2]
```diff
@@ -134,22 +121,22 @@ func markExperimentCompleted(instance *iter8v1alpha1.Experiment) {
...
 func successMsg(instance *iter8v1alpha1.Experiment) string {
 	if instance.Spec.Assessment == iter8v1alpha1.AssessmentOverrideSuccess {
-		return "OverrideSuccess"
+		return "Override Success"
 	} else if instance.Status.AssessmentSummary.AllSuccessCriteriaMet {
-		return "AllSuccessCriteriaMet"
+		return "All Success Criteria Were Met"
 	} else {
-		return "IterationsExhausted"
+		return "Last iteration was completed"
```
Comment (kalantar) [inline pkg/controller/experiment/common.go]: Picky observation: case does not match other return values.

Comment (kimikowang) [reply pkg/controller/experiment/common.go]: Thanks for pointing it out! Resolved.

**Comment by kimikowang** [comment]
Thanks for testing @kalantar :)

**Pull request merged**